src/main/java/org/generationcp/middleware/service/impl/study/StudyGermplasmServiceImpl.java

```diff
@@ -54,7 +59,8 @@ public StudyGermplasmServiceImpl(final HibernateSessionProvider sessionProvider)
 			studyGermplasmDto.setSeedSource(this.findStockPropValue(TermId.SEED_SOURCE.getId(), stockModel.getProperties()));
 			studyGermplasmDto.setCheckType(Integer.valueOf(this.findStockPropValue(TermId.ENTRY_TYPE.getId(), stockModel.getProperties())));
 			studyGermplasmDto.setStockIds(stockIdsMap.getOrDefault(stockModel.getGermplasm().getGid(), ""));
-			studyGermplasmDto.setGroupId(stockModel.getGermplasm().getMgid());
+			final String groupGid = this.findStockPropValue(TermId.GROUPGID.getId(), stockModel.getProperties());
+			studyGermplasmDto.setGroupId(groupGid != null ? Integer.valueOf(groupGid) : 0);
```

Comment:
Is it correct to set the groupGID value to 0 if the property is not present?

Reply:
I think so, because the default value of mgid is 0.